ddht/v5_1/rpc_handlers.py

```diff
@@ -115,8 +121,62 @@ async def do_call(self, params: FindNodesRPCParams) -> Tuple[str, ...]:
         return tuple(repr(enr) for enr in enrs)
 
 
+class GetENRHandler(RPCHandler[NodeID, GetENRResponse]):
+    def __init__(self, network: NetworkAPI) -> None:
+        self._network = network
+
+    def extract_params(self, request: RPCRequest) -> NodeID:
+        raw_params = extract_params(request)
+        validate_params_length(raw_params, 1)
+        raw_destination = raw_params[0]
+        node_id, _ = validate_and_extract_destination(raw_destination)
+        return node_id
+
+    async def do_call(self, params: NodeID) -> GetENRResponse:
+        response = self._network.enr_db.get_enr(params)
+        return GetENRResponse(enr_repr=repr(response))
+
+
+class SetENRHandler(RPCHandler[ENRAPI, None]):
+    def __init__(self, network: NetworkAPI) -> None:
+        self._network = network
+
+    def extract_params(self, request: RPCRequest) -> ENRAPI:
+        raw_params = extract_params(request)
+        validate_params_length(raw_params, 1)
+        enr_repr = raw_params[0]
+        try:
+            enr = ENR.from_repr(enr_repr)
+        except ValidationError:
+            raise RPCError(f"Invalid ENR repr: {enr_repr}")
+        return enr
+
+    async def do_call(self, params: ENRAPI) -> None:
+        self._network.enr_db.set_enr(params)
```

Comment:
Unhandled error here if the enr_db already has this record in the DB.  I think you can trigger this in tests by just sending in the same ENR twice.

Reply:
It looks to me that `enr_db` will only throw an error if the sequence number of the new enr is older than the existing enr [link](https://github.com/ethereum/eth-enr/blob/a9268bc16bf54881ee1e0c5245bb95f8aabe0176/eth_enr/enr_db.py#L49), so sending in the same enr wouldn't trigger this exception, though I could be missing something. I've added a test to catch the case with an outdated sequence number.